AmethystAPI/src/amethyst/runtime/patchManager/PatchManager.cpp

```diff
@@ -0,0 +1,68 @@
+#include "PatchManager.h"
+#include "../../Log.h"
+#include "../../Memory.h"
+namespace Amethyst {
+    // Returns false if the patch failed to apply
+    bool PatchManager::ApplyPatch(uintptr_t address, uint8_t* patch, size_t size){
+        DWORD oldProtection;
+        if (!UnprotectMemory(address, size, &oldProtection)) {
+            Log::Error("Failed to unprotect memory at address: 0x{0:x}", address);
+			return false;
+		}
+        Patch ptch;
+        auto it = this->m_Patches.find(address);
+        if (it != this->m_Patches.end()) [[unlikely]] {
+            if (memcmp(reinterpret_cast<void*>(it->second.address), patch, size)) {
+                Log::Warning("Patch already applied at address: 0x{0:x}", address);
+                ptch.original = it->second.original;
```

Comment:
We should probably apply both patches and not remove the first one.

What if the first patch had a larger size but same starting address, and then a second patch used that address but only did a smaller segment of it